### PR TITLE
fix(rlp): receipts from non legacy transactions returning wrong length

### PR DESCRIPTION
**Pull request by joshieDo**
As the title suggests. 

Found out by #829 , and according to @Rjected it should have shown when we calculate receipts root. Are we doing that yet? I'm guessing not.

### CODE REVIEW

**Review by Rjected — APPROVED** [review]
lgtm!

**Comment by onbjerg** [comment]
>Are we doing that yet? I'm guessing not.

Yes, we are

**Comment by rakita** [comment]
Before Byzantien recept type was different and had a state root at every receipt. We are currently skipping that:
https://github.com/paradigmxyz/reth/blob/e6ca4c56c6a314e5553656e5b62be3bca6801b85/crates/executor/src/executor.rs#L261-L268

Hash forcalculating receipt root is done over `01 | rlp(..)` and not `rlp(01 | rlp(..))` so this function is not called. We are doing receipt rot check in eth block tests.

**Review by gakonst — APPROVED** [review]
This is also present in the round-trip rlp PR I think

File: crates/primitives/src/receipt.rs [hunk 1]
```diff
@@ -112,7 +112,7 @@ impl Encodable for Receipt {
         if matches!(self.tx_type, TxType::EIP1559 | TxType::EIP2930) {
             payload_len += 1;
             // we include a string header for typed receipts, so include the length here
-            payload_len = length_of_length(payload_len);
+            payload_len += length_of_length(payload_len);
```
[the remark this answers is not in the document]
Comment (gakonst) [reply crates/primitives/src/receipt.rs:115]: The round-trip test above is that

**Comment by gakonst** [comment]
@joshieDo can merge pending conflicts

**Pull request merged**